### PR TITLE
chore(push notifications): enable disabling through server settings

### PR DESCRIPTION
**Pull request by digorath**
## Changes proposed in this PR:

Some communities do not immediately want to have push notifications enabled. so i added the option to disable it through server settings.
this way the permission popup and the settings should not show, when the server sided settings do explicitly set push notifications to a disabled state.

SVA-185

### CODE REVIEW

File: src/screens/SettingsScreen.js [hunk 1]
```diff
@@ -80,12 +80,15 @@ export const SettingsScreen = () => {
       });
 
     const updateSectionedData = async () => {
-      const { settings = { matomo: false } } = globalSettings;
-      const pushPermission = await readFromStore(PushNotificationStorageKeys.IN_APP_PERMISSION);
+      const { settings = { matomo: false }, pushNotifications } = globalSettings;
```
Comment (donni106) [inline src/screens/SettingsScreen.js:83]: do we want to have `matomo` nested in `settings` but `pushNotifications` not? why should there be a difference?

Comment (digorath) [reply src/screens/SettingsScreen.js]: updated it to be in line with other "on/off" features

File: src/hooks/PushNotification.ts [hunk 2]
```diff
@@ -16,8 +16,15 @@ type ResponseHandler = (arg: Notifications.NotificationResponse) => void;
 export const usePushNotifications = (
   notificationHandler?: NotificationHandler,
   interactionHandler?: ResponseHandler,
-  behavior?: Notifications.NotificationBehavior
+  behavior?: Notifications.NotificationBehavior,
+  active?: boolean
 ): void => {
+  // this causes the active state to never change between rerenders.
+  // like this enabling or disabling the pushNotifications requires an app restart.
```
Comment (donni106) [inline src/hooks/PushNotification.ts:23]: we should add the information on activation/deactivation like for matomo: `...wird [...] mit dem nächsten Neustart der App wirksam.`

Comment (digorath) [reply src/hooks/PushNotification.ts:23]: this is happening when the SERVER settings get changed.
do you think there is a reasonable way to notify a user about that?

Comment (donni106) [reply src/hooks/PushNotification.ts:23]: aha, haven't read it in a way, that the server is meant here.

**Review by donni106 — APPROVED** [review]

**Pull request merged**